src/cargo/ops/cargo_uninstall.rs

```diff
@@ -146,11 +146,13 @@ fn uninstall_pkgid(
         }
         tracker.remove(pkgid, &bins);
     }
-    tracker.save()?;
+
     for bin in to_remove {
         config.shell().status("Removing", bin.display())?;
         paths::remove_file(bin)?;
     }
 
+    // Only Save the tracker when remove Bin successfully.
+    tracker.save()?;
```

Comment:
This misses the following part of the suggested fix

> For paths::remove_file() and tracker.remove() should go hand in hand.

Or in other words
- The old behavior was to stop tracking binaries, leaking them
- The new approach is to remove binaries but we still track them

The proposed behavior in the issue is to keep the filesystem and the tracker in sync.

Reply:
> The new approach is to remove binaries but we still track them

If I'm not mistaken: the `the approach` is means what [this commit](09be0eef54a1e66d38012f1c1e7fd3db424e9938) had done.

Reply:
> The proposed behavior in the issue is to keep the filesystem and the tracker in sync.
I try to solving the problem in next commit